pyproject.toml

```diff
@@ -56,22 +62,31 @@ disallow_untyped_defs = true
 
 [[tool.mypy.overrides]]
 module = [
-  "setuptools.*",
-  "networkx.*",
-  "pygraphviz.*",
-  "cached_method.*",
-  "nose2.tools.*"
+    "setuptools.*",
+    "networkx.*",
+    "pygraphviz.*",
+    "cached_method.*",
+    "nose2.tools.*",
 ]
 ignore_missing_imports = true
 
-[tool.flake8]
-# Black compatibility
-max-line-length = 88
-extend-ignore = ["E203", "W503"]
+[tool.ruff]
+include = ["**/*.py"]
 
-# Per <https://black.readthedocs.io/en/stable/guides/using_black_with_other_tools.html#profile>
-[tool.isort]
-profile = "black"
+[tool.ruff.lint]
```

Comment:
This is a pretty minimal ruleset, but there are many others that can be enabled to check things like naming conventions, warnings against outdated programming patterns, etc: https://docs.astral.sh/ruff/rules/

Reply:
@eliotwrobson I think what you have is fine, as long as it generally covers what Flake8+black+isort already cover out of the box.

Reply:
Yup, it does that + more than what we had before.